### PR TITLE
Detect timeout and uri not found

### LINKED ISSUE #326: Add the ability to detect time-out
Add the ability to detect time-out as a separate and distinct event.

Currently timeouts are clumped together as a non 2xx error. 
It would be useful to detect timeout as a distinct event from other 500.

### PR DESCRIPTION
**Pull request by sapiderman**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. A lot of the errors previously are lumped at returned as a 500
2. This PR adds detection for timeut and not found
3. This PR closes #326 

## How did you implement / how did you fix it  
1.  Detect the events: "ECONNABORTED" and "ENOTFOUND" from axios

## How to test  
1. Run monika with some unknown uri and a quite timeout
2. Use the following config

  ```
  {
      "id": "httpbin1",
      "name": "probe httpbin",
      "requests": [
        {
          "method": "GET",
          "url": "https://xxxhttpbin.org/status/400",
          "saveBody": false,
          "timeout": 5000
        }
      ],
      "incidentThreshold": 3,
      "recoveryThreshold": 3
    },
    {
      "id": "httpbin2",
      "name": "probe httpbin",
      "requests": [
        {
          "method": "GET",
          "url": "https://httpbin.org/status/400",
          "saveBody": false,
          "timeout": 50
        }
      ],
      "incidentThreshold": 3,
      "recoveryThreshold": 3
    }
```
3. you should get something like:
```
WARN: 2021-08-30T03:31:33.680Z 1 id:httpbin1 0 GET https://xxxhttpbin.org/status/400 78ms, ALERT: URI not found, status-not-2xx
WARN: 2021-08-30T03:31:33.683Z 1 id:httpbin2 599 GET https://httpbin.org/status/400 53ms, ALERT: Request Timed out, status-not-2xx
```

### CODE REVIEW

File: src/plugins/validate-response/checkers/index.ts [hunk 1]
```diff
@@ -59,10 +59,21 @@ export const getResponseValue = (
...
+ * responseChecker checks the response against notif/alert triggers
+ * @param {obj} alert contains our alerts to look for
+ * @param {obj} res is the probe result
+ * @returns {boolean} flag true if we have something to alert/notify
+ */
 const responseChecker = (
   alert: ProbeAlert,
   res: AxiosResponseWithExtraData
 ): boolean => {
+  // if status is 599 : timeout or uri is not found (0), worth reporting so return true
```
Comment (gsukanto) [inline src/plugins/validate-response/checkers/index.ts:72]: If the timeout came from internet connection, will this report causing another problem?

**Review by gsukanto — APPROVED** [review]

File: src/plugins/validate-response/checkers/index.ts (hunk 1 above)
Comment (gsukanto) [inline src/plugins/validate-response/checkers/index.ts:72]: If the timeout came from internet connection, will this report causing another problem?

File: src/components/logger/index.ts [hunk 2]
```diff
@@ -38,20 +38,21 @@ const EventEmitter = getEventEmitter()
 
 /**
  * getStatusColor colorizes different statusCode
- * @param {number} statusCode is the httpStatus to colorize
+ * @param {any} responseCode is the httpStatus to colorize
  * @returns {string} color code based on chalk: Chalk & { supportsColor: ColorSupport };
  */
-export function getStatusColor(statusCode: number) {
-  switch (Math.trunc(statusCode / 100)) {
+export function getStatusColor(responseCode: any) {
```
Comment (kevinhermawan) [inline src/components/logger/index.ts:44]: Why is this type `any`?

Comment (sapiderman) [reply src/components/logger/index.ts:44]: thanks..good catch.  its back to :number

File: src/plugins/validate-response/checkers/index.ts (hunk 1 above)
Comment (sapiderman) [reply src/plugins/validate-response/checkers/index.ts:72]: this test was by disabling wifi while requesting... (hitting https://httpbin.org/status/200 with timout 5000)
still got consistent timout flag.
image

Comment (sapiderman) [reply src/plugins/validate-response/checkers/index.ts:72]: as the test above, the time out worked. thanks!

**Review by armedi — APPROVED** [review]

**Pull request merged**